test/cpp/qps/driver.cc

```diff
@@ -315,11 +317,18 @@ std::unique_ptr<ScenarioResult> RunScenario(
     if (!servers[i].stream->Read(&init_status)) {
       gpr_log(GPR_ERROR, "Server %zu did not yield initial status", i);
     }
-    gpr_join_host_port(&cli_target, host, init_status.port());
-    client_config.add_server_targets(cli_target);
-    gpr_free(host);
-    gpr_free(driver_port);
-    gpr_free(cli_target);
+    if (qps_server_target_override != NULL && strlen(qps_server_target_override) > 0) {
```

Comment:
Since the purpose of this seems to be to work around the gpr_join_host_port business, why not just have an option to disable that? That way you can still have multiple server targets.